bundles/org.palladiosimulator.dataflow.confidentiality.analysis/src/org/palladiosimulator/dataflow/confidentiality/analysis/sequence/CharacteristicsCalculator.java

```diff
@@ -1,5 +1,243 @@
 package org.palladiosimulator.dataflow.confidentiality.analysis.sequence;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.palladiosimulator.dataflow.confidentiality.analysis.PCMAnalysisUtils;
+import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.CharacteristicValue;
+import org.palladiosimulator.dataflow.confidentiality.analysis.sequence.entity.DataFlowVariable;
+import org.palladiosimulator.dataflow.confidentiality.pcm.model.confidentiality.ConfidentialityVariableCharacterisation;
+import org.palladiosimulator.dataflow.confidentiality.pcm.model.confidentiality.dictionary.DictionaryPackage;
+import org.palladiosimulator.dataflow.confidentiality.pcm.model.confidentiality.dictionary.PCMDataDictionary;
+import org.palladiosimulator.dataflow.confidentiality.pcm.model.confidentiality.expression.LhsEnumCharacteristicReference;
+import org.palladiosimulator.dataflow.confidentiality.pcm.model.confidentiality.expression.NamedEnumCharacteristicReference;
+import org.palladiosimulator.dataflow.dictionary.characterized.DataDictionaryCharacterized.EnumCharacteristicType;
+import org.palladiosimulator.dataflow.dictionary.characterized.DataDictionaryCharacterized.Literal;
+import org.palladiosimulator.dataflow.dictionary.characterized.DataDictionaryCharacterized.expressions.And;
+import org.palladiosimulator.dataflow.dictionary.characterized.DataDictionaryCharacterized.expressions.False;
+import org.palladiosimulator.dataflow.dictionary.characterized.DataDictionaryCharacterized.expressions.Or;
+import org.palladiosimulator.dataflow.dictionary.characterized.DataDictionaryCharacterized.expressions.Term;
+import org.palladiosimulator.dataflow.dictionary.characterized.DataDictionaryCharacterized.expressions.True;
+import org.palladiosimulator.pcm.parameter.VariableCharacterisation;
+
 public class CharacteristicsCalculator {
-    // FIXME: Remove if not needed anymore
+    private List<DataFlowVariable> currentVariables;
+
+    /**
+     * Initialize Characteristic Calculator with initial Variables
+     * 
+     * @param initialVariables
+     *            DataFlowVariables of the previous ActionSequence Element
+     */
+    public CharacteristicsCalculator(List<DataFlowVariable> initialVariables, List<CharacteristicValue> nodeVariables) {
+        this.currentVariables = new ArrayList<>(initialVariables);
+        DataFlowVariable nodeCharacteristicContainer = new DataFlowVariable("container");
+        nodeVariables.forEach(it -> nodeCharacteristicContainer.addCharacteristic(it));
+        currentVariables.add(nodeCharacteristicContainer);
+    }
+
+    /**
+     * Evaluate a Variable Characterization with the current Variables
```

Comment:
Please try to be a little more precise in the javadoc. Repeating the function name with minimal additional information is usually no helpful documentation